lychee-lib/src/filter/mod.rs

```diff
@@ -99,75 +100,74 @@ mod test {
     const V6_MAPPED_V4_PRIVATE_CLASS_A: &str = "http://[::ffff:10.0.0.1]";
     const V6_MAPPED_V4_LINK_LOCAL: &str = "http://[::ffff:169.254.0.1]";
 
+    fn get_host_v4(s: &str) -> Option<Ipv4Addr> {
```

Comment:
Given that `get_host_v4` and `get_host_v6` are only used in tests and not expected to fail, do you think it makes sense to just use `unwrap`? 
So instead of `Option<Ipv4Addr>`, we could just return `Ipv4Addr` directly.

Would make the call more simple:

```rust
get_host_v4(V4_PRIVATE_CLASS_A).ok_or(())?
```

becomes 

```rust
get_host_v4(V4_PRIVATE_CLASS_A)
```

Reply:
```rust
    macro_rules! assert_ip_address {
        (v4: $ip:expr, $predicate:tt) => {
            let res = if let Host::Ipv4(ipv4) = Url::parse($ip).map_err(|_| ())?.host().ok_or(())? {
                ipv4.$predicate()
            } else {
                false
            };
            std::assert!(res);
        };
        (v6: $ip:expr, $predicate:tt) => {
            let res = if let Host::Ipv6(ipv6) = Url::parse($ip).map_err(|_| ())?.host().ok_or(())? {
                ipv6.$predicate()
            } else {
                false
            };
            std::assert!(res);
        };
    }

    #[test]
    fn test_const_sanity() -> Result<(), ()> {
        assert_ip_address!(v4: V4_PRIVATE_CLASS_A, is_private);
        assert_ip_address!(v4: V4_PRIVATE_CLASS_B, is_private);
        assert_ip_address!(v4: V4_PRIVATE_CLASS_C, is_private);

        assert_ip_address!(v4: V4_LOOPBACK, is_loopback);
        assert_ip_address!(v6: V6_LOOPBACK, is_loopback);

        assert_ip_address!(v4: V4_LINK_LOCAL, is_link_local);

        Ok(())
    }
```

Reply:
Returning `Ipv4Addr` or `Ipv6Addr` without wrapping them in `Option` is error-prone. A small typo would break it.